### PR TITLE
Final functionality for Workbench Corrections Interface

### LINKED ISSUE #25828: Indirect Workbench Corrections - Deleted lines re-appear on preview plot
### Expected behavior
The blue line should not re-appear after the sample has been changed.

### Actual behavior
The blue line re-appears.

### Steps to reproduce the behavior
1. Workbench->`Indirect`->`Corrections`
1. Load the 26176 and 26174 data below 
2. Click Run and wait. The blue subtracted line should appear.
3. Load the other_red data below as sample. The blue subtracted line should disappear.
4. Change the plotted spectrum. The blue subtracted line reappears!

[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3246524/irs26176_graphite002_red.zip)
[irs26174_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3246526/irs26174_graphite002_red.zip)

[other_red.zip](https://github.com/mantidproject/mantid/files/3246527/other_red.zip)

### Platforms affected
All I assume

### LINKED ISSUE #25829: Indirect Workbench Corrections - Crash when loading new sample
### Expected behavior
No crash.

### Actual behavior
Crash.

### Steps to reproduce the behavior
1. Workbench->Indirect->Corrections
1. Load the other_red as sample
2. Load 26174 as the container
3. Load the 26176 as the sample
4. Wait for 5 seconds and it will crash.

**Data**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3246524/irs26176_graphite002_red.zip)
[irs26174_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3246526/irs26174_graphite002_red.zip)

[other_red.zip](https://github.com/mantidproject/mantid/files/3246527/other_red.zip)

### Platforms affected
All I assume. Definitely on windows.

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Fixes a few bugs with the Workbench version of Indirect Corrections. This also enables the plot functionality missing from #25715. 

**To test:**

Do the tests from #25715 and check that the bugs in the issues reference below are fixed along with verifying that the plot buttons now give the correct separate plot window.

Fixes #25829 
Fixes #25828 

*This does not require release notes* because **all of the indirect interfaces will be announced once they are completed.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/mplcpp/src/Axes.cpp [hunk 1]
```diff
@@ -79,6 +79,34 @@ void Axes::forEachArtist(const char *containerAttr, const ArtistOperation &op) {
...
+ * then this is a no-op.
+ * @param containerAttr The name of the container attribute
+ * @param label The label of the artists to remove
+ */
+void Axes::removeArtists(const char *containerAttr, const QString label) {
+  GlobalInterpreterLock lock;
+  const auto lineNameAsUnicode =
+      Python::NewRef(PyUnicode_FromString(label.toLatin1().constData()));
+  try {
+    auto container = pyobj().attr(containerAttr);
```
Comment (robertapplin) [inline qt/widgets/mplcpp/src/Axes.cpp]: Is `const auto` possible here?

File: qt/widgets/mplcpp/src/Plot.cpp [hunk 2]
```diff
@@ -60,6 +88,24 @@ constructKwargs(boost::optional<std::vector<int>> spectrumNums,
...
+
+Python::Object plot(const Python::Object &args,
+                    boost::optional<std::vector<int>> spectrumNums,
+                    boost::optional<std::vector<int>> wkspIndices,
+                    boost::optional<Python::Object> fig,
+                    boost::optional<QHash<QString, QVariant>> plotKwargs,
+                    boost::optional<QHash<QString, QVariant>> axProperties,
+                    boost::optional<std::string> windowTitle, bool errors,
+                    bool overplot) {
+  auto kwargs = constructKwargs(spectrumNums, wkspIndices, fig, plotKwargs,
```
Comment (robertapplin) [inline qt/widgets/mplcpp/src/Plot.cpp]: `const auto`?

File: qt/widgets/mplcpp/test/PlotTest.h [hunk 3]
```diff
@@ -32,37 +33,37 @@ class PlotTest : public CxxTest::TestSuite {
...
-    alg->execute();
-
     setMatplotlibBackend();
+    createTestWorkspaceInADS(m_testws_name);
   }
 
   void testPlottingWorksWithWorkspaceIndex() {
-    std::vector<std::string> workspaces = {"ws"};
+    std::vector<std::string> workspaces = {m_testws_name};
+    std::vector<int> index = {1};
```
Comment (robertapplin) [inline qt/widgets/mplcpp/test/PlotTest.h]: `const`'s in these tests?

File: qt/scientific_interfaces/Indirect/IndirectCorrections.cpp [hunk 4]
```diff
@@ -72,6 +72,7 @@ void IndirectCorrections::initLayout() {
             SLOT(showMessageBox(const QString &)));
   }
 
+  m_uiForm.pbSettings->setIcon(IndirectSettings::icon());
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/IndirectCorrections.cpp]: Thanks for doing this, this was on my list! Could you add the same line to `IndirectDiffractionReduction.cpp`?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Good job, the external plotting of Spectra and contour plots work well. The bugs are also fixed. Just a few picky `const` comments.

File: qt/widgets/mplcpp/src/Axes.cpp (hunk 1 above)
Comment (martyngigg) [reply qt/widgets/mplcpp/src/Axes.cpp]: Thanks for the `const` catches!

**Review by robertapplin — DISMISSED** [review]

**Review by robertapplin — APPROVED** [review]

**Comment by martyngigg** [comment]
@mantidproject/gatekeepers I'd appreciate it if this could get into nightly tonight.

File: qt/widgets/mplcpp/inc/MantidQtWidgets/MplCpp/Plot.h [hunk 5]
```diff
@@ -21,9 +22,16 @@ namespace MantidQt {
 namespace Widgets {
 namespace MplCpp {
 
+/// C++ counterpart to MantidAxType in mantid.plots.utility
```
Comment (DanNixon) [inline qt/widgets/mplcpp/inc/MantidQtWidgets/MplCpp/Plot.h]: Minor point, this should be a Doxygen style comment.

**Pull request merged**